src/bootstrap/lib.rs

```diff
@@ -607,8 +607,15 @@ impl Build {
 
     /// Get the space-separated set of activated features for the standard
     /// library.
-    fn std_features(&self) -> String {
+    fn std_features(&self, compiler: &Compiler) -> String {
         let mut features = "panic-unwind".to_string();
+
+        // The sanitizer runtimes depend on LLVM, which gets build during stage0
+        // rustc
+        if compiler.stage != 0 {
+            features.push_str(" asan lsan msan tsan")
+        }
```

Comment:
I've run into lots of problems historically where if we change features in each stage it causes lots of recompilations to happen that wouldn't happen otherwise. I *think* it's related to cross-compilation and such, but I forget precisely what's happening here. 

Is there perhaps another way we can communicate this information without switching features on each compiler?